### PR TITLE
if UseEmulation is configured, checking kvm is non-essential

### PR DESCRIPTION
**Pull request by lixuehui08**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by vladikr** [comment]
@lixuehui08 thank you for this PR!

Could you please add more content to the PR description?

Also, please make sure to add "Signed-off-by" to your commit message, as it's required by the project.
https://github.com/kubevirt/kubevirt/blob/master/CONTRIBUTING.md#contributor-compliance-with-developer-certificate-of-origin-dco

File: pkg/virt-launcher/virtwrap/api/converter.go [hunk 1]
```diff
@@ -898,16 +898,15 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
...
-	if _, err := os.Stat("/dev/kvm"); os.IsNotExist(err) {
-		if c.UseEmulation {
-			logger := log.DefaultLogger()
-			logger.Infof("Hardware emulation device '/dev/kvm' not present. Using software emulation.")
-			domain.Spec.Type = "qemu"
-		} else {
+	if c.UseEmulation {
+		logger := log.DefaultLogger()
+		logger.Infof("Using software emulation.")
+	} else {
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/api/converter.go:904]:
```suggestion
        domain.Spec.Type = "qemu"
	} else {
```

**Review by xpivarc — CHANGES_REQUESTED** [review]
Thanks for the PR, @lixuehui08 .

**Comment by rmohr** [comment]
@lixuehui08 could you explain in the PR descrition what you try to fix?

**Comment by lixuehui08** [comment]
Tanks for your request，in view of this problem, there are two aspects to consider：




Firstly，checking if the kvm device exists is redundant when i want to use software emulation，and it will take time and resource.




Secondly,checking kvm device every time and it's log confused me and made me feel like my configuration wasn't working.




Considering the above considerations，i want to optimize it and look forward to our products getting better and better.




Tanks again for your attention.

 

-----原始邮件-----
发件人:"Roman Mohr" <notifications@github.com>
发送时间:2020-12-04 20:11:49 (星期五)
收件人: kubevirt/kubevirt <kubevirt@noreply.github.com>
抄送: lixuehui08 <lixuehui_08@zju.edu.cn>, Mention <mention@noreply.github.com>
主题: Re: [kubevirt/kubevirt] if UseEmulation is configured, checking kvm is non-essential (#4618)






@lixuehui08 could you explain in the PR descrition what you try to fix?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

78l

**Comment by rmohr** [comment]
> Firstly，checking if the kvm device exists is redundant when i want to use software emulation

Let me clarify this point: We only support kubevirt with kvm. You can only configure to allow a fallback to kvm. It should first check if kvm can be used. That part is intentional.

> and it will take time and resource. 

Hm, the kvm device can appear and disappear. We can probably base our check somehow on the deviceplugin which uses  inotify to check that. I think the check is done at the wrong place. The converter should not check anything, it should only get given inputs.

> Secondly,checking kvm device every time and it's log confused me and made me feel like my configuration wasn't working. Considering the above considerations，

I can really understand that this is annoying. This may be one of the main problems.


> i want to optimize it and look forward to our products getting better and better.

:+1:

**Comment by lixuehui08** [comment]
Hi,




I'm sorry to disturb you again，cloud i confirm one question with you?




I want to confirm if kubevirt suuport injecting password through cloudbase-init for windows virtual machine, because there is less information on this aspect and i injected in failure.




Thanks，looking forward to your reply 



-----原始邮件-----
发件人:"Roman Mohr" <notifications@github.com>
发送时间:2020-12-04 21:56:38 (星期五)
收件人: kubevirt/kubevirt <kubevirt@noreply.github.com>
抄送: lixuehui08 <lixuehui_08@zju.edu.cn>, Mention <mention@noreply.github.com>
主题: Re: [kubevirt/kubevirt] if UseEmulation is configured, checking kvm is non-essential (#4618)






Firstly，checking if the kvm device exists is redundant when i want to use software emulation

Let me clarify this point: We only support kubevirt with kvm. You can only configure to allow a fallback to kvm. It should first check if kvm can be used. That part is intentional.

and it will take time and resource.

Hm, the kvm device can appear and disappear. We can probably base our check somehow on the deviceplugin which uses inotify to check that. I think the check is done at the wrong place. The converter should not check anything, it should only get given inputs.

Secondly,checking kvm device every time and it's log confused me and made me feel like my configuration wasn't working. Considering the above considerations，

I can really understand that this is annoying. This may be one of the main problems.

i want to optimize it and look forward to our products getting better and better.

👍

**Pull request closed without merging**